tests/group_functions/type_coverage.h

```diff
@@ -0,0 +1,708 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provide common functions for type coverage
+//
+*******************************************************************************/
+
+/**
+ * FIXME: this is an extended (and also partial) copy of type_coverage.h
+ * with marray parts removed for hipSYCL compilation
+ * and with for_all_combinations_with/for_all_combinations_with_two added
+ */
+
+#ifndef __SYCLCTS_TESTS_COMMON_TYPE_COVERAGE_H
+#define __SYCLCTS_TESTS_COMMON_TYPE_COVERAGE_H
+
+#include <optional>
+#include <string>
+#include <utility>
+#include <variant>
+
+#include <sycl/sycl.hpp>
+
+#include "../../util/type_traits.h"
+
+#include "catch2/catch_tostring.hpp"
+
+#include <cstddef>  // for std::size_t
+
+/**
+ * @brief Retrieve type name; by default just forward the given one
+ */
+template <typename T>
+struct type_name_string {
+  static std::string get(std::string dataType) { return dataType; }
+};
+
+/**
+ * @brief Specialization of type name retrieve for sycl::vec class
+ * @param T Type of the data stored in vector
+ * @param nElements Number of elements stored in vector
+ */
+template <typename T, size_t nElements>
```

Comment:
I wonder whether we could not just add https://github.com/Neargye/nameof as a dependency of this project instead of doing all of this. :-)